libmachine/provision/ubuntu_systemd.go

```diff
@@ -102,6 +102,19 @@ func (provisioner *UbuntuSystemdProvisioner) Provision(swarmOptions swarm.Option
 	provisioner.EngineOptions = engineOptions
 	swarmOptions.Env = engineOptions.Env
 
+	log.Debug("waiting for cloud-init to complete")
+	err := mcnutils.WaitFor(func() bool {
```

Comment:
From what I understand this retries up to 60 times with delay of 3 seconds between retries which sounds a bit too much. Is this the standard practice we already follow in machine code when waiting for operations like this?

Reply:
> To get some more context, what was the number of iterations the wait went through in your testing?

I ended up having to wait 3-5 times when just running `apt-get update` - theoretically this should just be one though. I left the retry in initially just due to the fact that I _really_ wanted to make sure it was init'd before continuing since other steps are paractically bound to fail if it isn't completed.

> And to add more to this, how long does actual command sudo cloud-init status --wait waits for? As per the code linked in another comment it looks like the retry logic waits for command function to complete and doesn't terminate it with any reasonable timeout.

so it [looks like](https://cloudinit.readthedocs.io/en/latest/explanation/boot.html#final) the command waits indefinitely which is nice - I changed it to just a helper function instead of retrying.  

I almost wonder if we should put an artificial timeout on it though just in case - something really long like 5m? Though I guess rancher itself will retry the node if the provision takes too long so maybe that is good enough.